design-proposals/golden-image-delivery-and-update-pipeline.md

```diff
@@ -0,0 +1,275 @@
+
+This means if a template references a PVC directly within the DataVolumeTemplates, then the UI should maintain the existing behavior of being able to display that the boot source is available if the PVC exists. In addition to this, if a template references a DataSource within the DataVolumeTemplates, the UI must be able to display the boot source is available if the DataSource exists and has the ready condition set to true.
+
+## Implementation Phases
+
+**Phase1** - CDI APIs
+* DataSource API/controller
+* DataImportCron API/controller
+
+**Phase2** - SSP + Common Templates
+* Common Templates transition to DataSources for DataVolumeTemplate source
+* SSP Operator creates DataImportCrons for automatic PVC import
+
+**Phase3** - HCO and UI
+* HCO Team integrates with SSP Operator’s API for setting volume import sources
+* UI Team transitions to detecting boot source using a linked DataSource rather than looking directly for a PVC of a specific name.
+
+# API Examples
+## VM + DataVolumeTemplate + DataSource as source
+
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataVolume
+metadata:
+   name: my-fedora-vm-image
+spec:
+  pvc:
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: 2Gi
+  dataSource:
+    name: fedora
+    namespace: golden-images 
+```
+
+## DataImportCron Example
+
+This example polls for image tags for the quay.io/kubevirt/fedora image and imports the latest image sorting tags based on semver. As new tags are introduced, the DataImportCron will import them as new PVCs and automatically manage updating the corresponding DataSource.
+
+* **source** specifies where to poll from
+* **managedDataSource** specifies the name of the corresponding DataSource this cron will manage. The controller should create this DataSource if it doesn't exist, and adopt ownership of the DataSource if it does already exist.
+* **schedule** specifies in cron format when and how often to look for new imports 
+* **garbageCollectOutdated** specifies whether old PVCs should be cleaned up after a new PVC is imported.
+
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataImportCron
+metadata:
+  name: fedora-image-import-cron
+  namespace: golden-images
+spec:
+  source:
+    registry:
+      url: docker://quay.io/kubevirt/fedora
+  schedule: "0 */12 * * *"
+  garbageCollectOutdated: true
+  managedDataSource: fedora
+status:
+  lastImportedPVC: quay.io/kubevirt/fedora:33
+```
+
+## DataSource Example
+
+This example shows how a DataSource uses a ready condition to indicate that the corresponding pvc exists and is populated. 
+
+The UI can use the ready condition to determine if a boot source exists for a DataSource or not. 
+
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataSource
+metadata:
+  name: fedora
+spec:
+  source:
+    pvc:
+      namespace: golden-images 
+      name: fedora-32-001
+status:
+  conditions:
+  - status: "True"
```

Comment:
Wondering where this is set.  Probably have to have a dedicated controller just for keeping this in sync.  Shouldn't be a big deal though

Reply:
yep, that's what I was thinking. hopefully it's pretty trivial